rust/trace/src/lib.rs

```diff
@@ -819,6 +821,22 @@ impl Trace {
         samples.sort_unstable();
         samples
     }
+
+    pub fn save<P: AsRef<Path>>(
+        &self,
+        path: P,
+        sort: bool,
+    ) -> Result<(), chrome_trace_dump::Error> {
+        let traces = if sort { samples_cloned_sorted() } else { samples_cloned_unsorted() };
+
+        if fs::metadata(&path)?.is_file() {
```

Comment:
what if it's a directory?

In anycase I would just suggest `if path.exists() { .. }`.

Also, in these fns where we take `AsRef<Path>`, the first thing I normally do is `let path = path.into()` so that we are working concretely with a `&Path`, and can use `Path` fns.